### PR TITLE
fix(canal): fix mysql unsigned medium int

### PR DESCRIPTION
**Pull request by Ryan-Git**
this is port from https://github.com/moiot/gravity/issues/182.

the problem is we use int32 to receive mysql medium int value, which is 3 bytes. When it's unsigned, simply do a type cast is not sufficient.

### CODE REVIEW

File: canal/rows.go [hunk 1]
```diff
@@ -57,7 +58,14 @@ func (r *RowsEvent) handleUnsigned() {
 			case int16:
 				r.Rows[i][index] = uint16(t)
 			case int32:
-				r.Rows[i][index] = uint32(t)
+				if strings.Contains(strings.ToLower(r.Table.Columns[i].RawType), "mediumint") {
```
Comment (siddontang) [inline canal/rows.go]: interesting, can you add some comments here? 

maybe it is better to add a test.

Comment (NearTan) [reply canal/rows.go]: Maybe the implementation from [gh-ost](https://github.com/github/gh-ost/blob/d2726c77f86cb65e25bce5c0ac5fe3fa0c997488/go/sql/types.go#L57-L67) is better and more clear?

Comment (Ryan-Git) [reply canal/rows.go]: @siddontang added comments and test. 
@NearTan I think so, thanks.

Comment (siddontang) [reply canal/rows.go]: Maybe we can do more, to add a similar MediumIntColumnType.
I worry that comparing string every time can reduce the performance.

Comment (Ryan-Git) [reply canal/rows.go]: good idea

**Comment by siddontang** [comment]
CI failed

**Comment by Ryan-Git** [comment]
fixed

**Pull request merged**